### PR TITLE
Add tox-py39 job

### PR DESCRIPTION
**Pull request by coreycb**

### CODE REVIEW

**Review by ChrisMacNaughton — APPROVED** [review]
LGTM, with the caveat that it can't actually run until we have an impish-medium node defined

**Review by freyes — APPROVED** [review]
LGTM. Corey already execute `juju config nodepool nodepool_config` to include impish, so probably this is good to be merged.

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -79,6 +79,21 @@
...
+
+      Uses tox with the ``py39`` environment.
+
+    nodeset:
+      nodes:
+        - name: impish-medium
+          label: impish-medium
+    vars:
+      tox_envlist: py39
+      python_version: 3.9
```
Comment (ajkavanagh) [inline zuul.d/jobs.yaml:96]: it may need to be "3.9" (quoted) as opposed to 3.9 as that might come through as a float in yaml-speak.

Comment (freyes) [reply zuul.d/jobs.yaml:96]: `python_version: 3.5` in the previous definition is not quoted, so I assumed the parser was going to do the right thing :thinking:

**Pull request merged**